test/test_python_dispatch.py

```diff
@@ -447,5 +447,24 @@ def wrap(e):
         self.assertEqual(x.grad, y)
         self.assertEqual(y.grad, x)
 
+    def test_detach_dispatched_for_non_subclassed_tensor(self) -> None:
+        num_detach_dispatched = 0
+
+        class FakeTensor(torch.Tensor):
+            @classmethod
+            def __torch_dispatch__(cls, func, types, args, kwargs):
+                nonlocal num_detach_dispatched
+
+                if func == torch.ops.aten.detach:
+                    num_detach_dispatched += 1
+
+                with no_dispatch():
+                    return func(*args, **kwargs)
+
+        with enable_python_mode(FakeTensor):
+            tmp = torch.ones([10, 10]).detach()
+
+        self.assertEqual(num_detach_dispatched, 2)
```

Comment:
@albanD I always thought it was weird that Tensor.detach() shows up in `__torch_dispatch__`. If we are marketing `__torch_dispatch__` as allowing users to override "the PyTorch API surface after autograd", then shouldn't detach not be a part of that API?

Reply:
I was having this exact discussion with Jeffrey on the forward AD functions like make/unpack dual.
I think that once we have pre dispatch extension they should not indeed. The current implementation is hacky to be able to catch it even though it shouldn't show up.

Note that the way to do this is simply to implement the detach function without a redispatch at the autograd level.